diff -rupP xkeyboard-config-2.0.orig/rules/HDR xkeyboard-config-2.0/rules/HDR
--- xkeyboard-config-2.0.orig/rules/HDR	2016-06-28 19:31:02.814647638 +0900
+++ xkeyboard-config-2.0/rules/HDR	2016-06-28 19:33:58.251517616 +0900
@@ -15,6 +15,7 @@
 ! model		layout[3]	variant[3]	=	symbols
 ! model		layout[4]	variant[4]	=	symbols
 ! model		=	symbols
+! model		layout		=	symbols
 ! layout	variant		=	compat
 ! layout[1]	variant[1]	=	compat
 ! layout[2]	variant[2]	=	compat
diff -rupP xkeyboard-config-2.0.orig/rules/base.ml_s1.part xkeyboard-config-2.0/rules/base.ml_s1.part
--- xkeyboard-config-2.0.orig/rules/base.ml_s1.part	1970-01-01 09:00:00.000000000 +0900
+++ xkeyboard-config-2.0/rules/base.ml_s1.part	2016-06-28 19:35:02.979838710 +0900
@@ -0,0 +1 @@
+ $inetmediakbds	jp	=	+jp(henkan)
diff -rupP xkeyboard-config-2.0.orig/symbols/jp xkeyboard-config-2.0/symbols/jp
--- xkeyboard-config-2.0.orig/symbols/jp	2016-06-28 19:31:03.046648788 +0900
+++ xkeyboard-config-2.0/symbols/jp	2016-06-28 19:32:57.367215702 +0900
@@ -105,6 +105,14 @@ xkb_symbols "common" {
     };
 };
 
+partial alphanumeric_keys
+xkb_symbols "henkan" {
+    key <XFER> {
+        type[Group1]= "PC_ALT_LEVEL2",
+        symbols[Group1]= [ Henkan, Mode_switch ]
+    };
+};
+
 // OADG109A map
 partial alphanumeric_keys
 xkb_symbols "OADG109A" {
